crates/compiler/can/src/desugar.rs

```diff
@@ -671,14 +684,14 @@ pub fn desugar_expr<'a>(
                 ),
             });
 
-            if field_names.len() == 2 {
+            if field_data.len() == 2 {
                 return arena.alloc(Loc {
                     region: loc_expr.region,
                     value: Apply(
                         new_mapper,
                         arena.alloc_slice_copy(&[
-                            field_vals[0],
-                            field_vals[1],
+                            field_data[0].1,
```

Comment:
minor nit: all this tuple indexing is a little hard to read. I'd prefer to define a struct under this scope so the fields are named.